nodestream_github/audit.py

```diff
@@ -0,0 +1,47 @@
+"""
+Nodestream Extractor that extracts audit logs from the GitHub REST API.
+
+Developed using Enterprise Server 3.12
+https://docs.github.com/en/enterprise-server@3.12/rest?apiVersion=2022-11-28
+"""
+
+from collections.abc import AsyncGenerator
+
+from nodestream.pipeline import Extractor
+
+from .client import GithubRestApiClient
+from .logging import get_plugin_logger
+from .types import GithubAuditLog
+
+logger = get_plugin_logger(__name__)
+
+
+class GithubAuditLogExtractor(Extractor):
+    """
+    Extracts audit logs from the GitHub REST API.
+    You can pass the enterprise_name, actions and lookback_period to the extractor
+    along with the regular GitHub parameters.
+
+    lookback_period can contain keys for days, months, and/or years as ints
+    actions can be found in the GitHub documentation
+    https://docs.github.com/en/enterprise-cloud@latest/admin/monitoring-activity-in-your-enterprise/reviewing-audit-logs-for-your-enterprise/searching-the-audit-log-for-your-enterprise#search-based-on-the-action-performed
```

Comment:
Should this link go to 3.12's docs, to match the module comment?

https://docs.github.com/en/enterprise-server@3.12/admin/monitoring-activity-in-your-enterprise/reviewing-audit-logs-for-your-enterprise/searching-the-audit-log-for-your-enterprise#searching-the-audit-log